src/keyword_vpd_parser.cpp

```diff
@@ -0,0 +1,122 @@
+#include "keyword_vpd_parser.hpp"
+
+#include "constants.hpp"
+#include "exceptions.hpp"
+
+#include <iostream>
+#include <numeric>
+#include <string>
+
+namespace vpd
+{
+
+/* @brief Keyword VPD parser function*/
+std::variant<types::ParsedVPD, types::KeywordVpdMap> KeywordVpdParser::parse()
+{
+    m_vpdIterator = m_keywordVpdVector.begin();
+
+    checkNextBytesValidity(sizeof(constants::KW_VPD_START_TAG));
+    std::advance(m_vpdIterator, sizeof(constants::KW_VPD_START_TAG));
+
+    size_t dataSize = getKwDataSize();
+
+    checkNextBytesValidity(constants::TWO_BYTES + dataSize);
+
+    // +TWO_BYTES is the size byte
+    std::advance(m_vpdIterator, constants::TWO_BYTES + dataSize);
```

Comment:
Why these checks were moved to this function ( line 16 to 26) it was there in separate function which was more readable.